src/errors.rs

```diff
@@ -1,26 +1,21 @@
-use ethers::prelude::{AbiError, ContractError};
-use ethers::providers::{Middleware, ProviderError};
-use ethers::types::{H160, U256};
+use alloy::primitives::{Address, U256};
+use alloy::transports::TransportError;
+
 use std::time::SystemTimeError;
 use thiserror::Error;
 use tokio::task::JoinError;
 use uniswap_v3_math::error::UniswapV3MathError;
 
 #[derive(Error, Debug)]
-pub enum AMMError<M>
-where
-    M: Middleware,
-{
-    #[error("Middleware error")]
-    MiddlewareError(<M as Middleware>::Error),
-    #[error("Provider error")]
-    ProviderError(#[from] ProviderError),
+pub enum AMMError {
+    #[error("Transport error")]
```

Comment:
Lets update all of error descriptions to `#[error(transparent)]` where we use `#[from] OtherError`.